proxy/fastcgi/agent.go

```diff
@@ -0,0 +1,116 @@
+package fastcgi
+
+import (
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/fabiolb/fabio/config"
+	"github.com/fabiolb/fabio/logger"
+	"github.com/fabiolb/fabio/metrics"
+	"github.com/fabiolb/fabio/noroute"
+	"github.com/fabiolb/fabio/proxy/gzip"
+	"github.com/fabiolb/fabio/route"
+	"github.com/fabiolb/fabio/uuid"
+)
+
+// Agent is a reverse proxy for FastCGI protocol
+type Agent struct {
```

Comment:
I think you don't need this and you should just add another option here:

https://github.com/fabiolb/fabio/blob/master/proxy/http_proxy.go#L149-L168